### PR TITLE
Chore/simplify ip retrieval

### PR DESCRIPTION
**Pull request by rahulghangas**
Removes the need to add and delete ip address from table. There are two primary reasons for not having the current usage of ip addresses with the peer table
- Constant need to interact with hash table with each connection (calculating hashes is comparatively expensive than just passing pointers)
- the addition and deletion of an ip address is done in two different layers (`transport` and `peer` respectively). This is harder to maintain and can lead to bugs

Rather, we define a new struct `Packet` in the `wire` package that includes both the message and the ip address. The inbound channel queue now accepts a `wire.Packet` instead of a `wire.Msg`, from which we can extract the message and the ip address and pass it as an additional argument to the receive callback function

### CODE REVIEW

**Review by ross-pure — COMMENTED** [review]
A question before continuing the review: is there any reason why the `wire.Msg` type is only replaced with the `wire.Packet` type in a few places? I notice there are quite a few changes that change the interface of a function to instead of only take a `wire.Msg` to take both a `wire.Msg` and a `net.Addr`, whereas I would have expected instead the change to be simply to make it take a `wire.Packet`.

**Comment by rahulghangas** [comment]
No particular reason, although just using `wire.Packet` would make be better. I'll update it

File: peer/peer.go [hunk 1]
```diff
@@ -87,23 +87,23 @@ func (p *Peer) DiscoverPeers(ctx context.Context) {
 }
 
 func (p *Peer) Run(ctx context.Context) {
-	p.transport.Receive(ctx, func(from id.Signatory, msg wire.Msg) error {
+	p.transport.Receive(ctx, func(from id.Signatory, packet wire.Packet) error {
 		// TODO(ross): Think about merging the syncer and the gossiper.
-		if err := p.syncer.DidReceiveMessage(from, msg); err != nil {
+		if err := p.syncer.DidReceiveMessage(from, packet); err != nil {
```
Comment (ross-pure) [inline peer/peer.go:92]: The syncer doesn't use the `IPAddr` field of the `Packet`, so the `DidReceiveMessage` function interface doesn't need to change. You may wish to keep the change for consistency, however. I would err on the side of reverting the change.

File: peer/peer.go [hunk 2]
```diff
@@ -87,23 +87,23 @@ func (p *Peer) DiscoverPeers(ctx context.Context) {
...
 			return err
 		}
-		if err := p.gossiper.DidReceiveMessage(from, msg); err != nil {
+		if err := p.gossiper.DidReceiveMessage(from, packet); err != nil {
```
Comment (ross-pure) [inline peer/peer.go:95]: Same as the comment above about the syncer interface.

File: transport/transport_test.go [hunk 3]
```diff
@@ -1,9 +1,6 @@
 package transport_test
```
Comment (ross-pure) [inline transport/transport_test.go:1]: With the deprecated test removed, nothing in this file is used any more (in particular, `setup`). We can keep it if you think it might get used at a later stage. In any case, this might suggest that we need to create some transport tests now that there are none.

**Review by ross-pure — APPROVED** [review]
Looks mostly good to me. I'm approving but I've left some suggestions that you might like to consider.

**Review by ross-pure — APPROVED** [review]
LGTM
